### PR TITLE
feat: add slot to custom description when data table is empty

### LINKED ISSUE #86: Allow customisation of empty data table content
## What problem does the feature solve?

If an error occurs while loading the data used to fill the data table, I would like to show an error message rather than just saying "No data".

## What does the proposed API look like?

Other frameworks tend to handle this with a slot.

### PR DESCRIPTION
**Pull request by zhypower1997**
fix #86

### CODE REVIEW

File: CHANGELOG.en-US.md [hunk 1]
```diff
@@ -808,3 +808,7 @@ See vue3.md
 ### Fixes
 
 - Rails of `n-scrollbar` shadow mouse event.
+
+### Features
+
+- `n-date-table` add `empty-description` slot.
```
Comment (07akioni) [inline CHANGELOG.en-US.md:814]: `empty` slot is okay.

File: CHANGELOG.zh-CN.md [hunk 2]
```diff
@@ -810,3 +810,7 @@
 ### Fixes
 
 - `n-scrollbar` 的轨道会挡住鼠标事件
+
+### Features
+
+- `n-data-table` 增加了 `'empty-description'`插槽
```
Comment (07akioni) [inline CHANGELOG.zh-CN.md:816]:
```suggestion
- `n-data-table` 增加了 empty 插槽
```

File: src/data-table/demos/enUS/index.demo-entry.md [hunk 3]
```diff
@@ -84,6 +84,14 @@ These methods can help you control table in an uncontrolled manner. However, it'
 | page | `(page: number) => void` |  |
 | sort | `(columnKey: string \| number \| null, order: 'ascend' \| 'descend' \| false) => void` | If columnKey set to `null`, it is the same as clearSorter. |
 
+## Slots
+
+### Data Table Slots
```
Comment (07akioni) [inline src/data-table/demos/enUS/index.demo-entry.md:89]:
```suggestion
### Slots
```

File: src/data-table/demos/enUS/index.demo-entry.md [hunk 4]
```diff
@@ -84,6 +84,14 @@ These methods can help you control table in an uncontrolled manner. However, it'
...
+
+| 名称  | 参数 | 说明                                           |
```
Comment (07akioni) [inline src/data-table/demos/enUS/index.demo-entry.md:91]: English header.

File: src/data-table/demos/zhCN/index.demo-entry.md [hunk 5]
```diff
@@ -84,6 +84,14 @@ tree
 | page | `(page: number) => void` |  |
 | sort | `(columnKey: string \| number \| null, order: 'ascend' \| 'descend' \| false) => void` | 如果 columnKey 设为 `null`，那它和 clearSorter 效果一致 |
 
+## Slots
+
+### Data Table Slots
```
Comment (07akioni) [inline src/data-table/demos/zhCN/index.demo-entry.md:89]:
```suggestion
### Slots
```

File: CHANGELOG.en-US.md (hunk 1 above)
Comment (zhypower1997) [reply CHANGELOG.en-US.md:814]: okay~

File: CHANGELOG.zh-CN.md [hunk 6]
```diff
@@ -810,3 +810,7 @@
 ### Fixes
 
 - `n-scrollbar` 的轨道会挡住鼠标事件
+
+### Features
+
+- `n-data-table` 增加了 `'empty'`插槽
```
Comment (07akioni) [inline CHANGELOG.zh-CN.md:816]:
```suggestion
- `n-data-table` 增加了 empty 插槽
```

File: src/data-table/demos/enUS/index.demo-entry.md [hunk 7]
```diff
@@ -84,6 +84,14 @@ These methods can help you control table in an uncontrolled manner. However, it'
 | page | `(page: number) => void` |  |
 | sort | `(columnKey: string \| number \| null, order: 'ascend' \| 'descend' \| false) => void` | If columnKey set to `null`, it is the same as clearSorter. |
 
+## Slots
+
+### Slots
+
+| Name  | Type | Description                                    |
+| ----- | ---- | ---------------------------------------------- |
+| empty | `()` | Custom description when data of table is empty |
```
Comment (07akioni) [inline src/data-table/demos/enUS/index.demo-entry.md:93]:
```suggestion
| empty | `()` | Custom description when data of table is empty. |
```

File: src/data-table/tests/DataTable.spec.tsx [hunk 8]
```diff
@@ -8,6 +8,38 @@ describe('n-data-table', () => {
   it('should work with import on demand', () => {
     mount(NDataTable)
   })
+  it('show custom empty', () => {
+    const columns = [
```
Comment (07akioni) [inline src/data-table/tests/DataTable.spec.tsx:12]: make columns simpler.

File: src/data-table/tests/DataTable.spec.tsx [hunk 9]
```diff
@@ -8,6 +8,38 @@ describe('n-data-table', () => {
...
+        title: 'Tags',
+        key: 'tags'
+      },
+      {
+        title: 'Action',
+        key: 'actions'
+      }
+    ]
+    const data: any = []
+    mount(() => (
```
Comment (07akioni) [inline src/data-table/tests/DataTable.spec.tsx:35]: You need to assert something. For example `expect(wrapper.find(...).exists()).toEqual(true)`

**Review by 07akioni — APPROVED** [review]

**Pull request merged**